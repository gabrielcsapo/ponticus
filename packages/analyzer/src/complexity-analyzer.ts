"use strict";

import { sum } from "lodash";

// @ts-ignore
import ESComplex from "typhonjs-escomplex";

export function analyse(js: string) {
    return ESComplex.analyzeModule(js);
}

export function process(analyses: any) {
    const summary = ESComplex.processProject(analyses);

    summary.totalLOC = sum(summary.reports.map((report: any) => report.aggregate.sloc.logical));

    return summary;
}