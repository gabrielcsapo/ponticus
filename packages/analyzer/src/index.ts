export { analyse, process } from './complexity-analyzer';