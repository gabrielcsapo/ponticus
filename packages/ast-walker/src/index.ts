import { type PluginEvent } from "@ponticus/plugin-manager";

import ASTWalker from "@ponticus/ast-walker/ASTWalker";

export { ASTWalker };

/**
 * Default walker instance.
 */
const walker = new ASTWalker();

/**
 * Provides a default ASTWalker instance.
 */
export default walker;

/**
 * Wires up walker on the plugin eventbus.
 *
 * @param ev - The plugin event.
 *
 */
export function onPluginLoad(ev: PluginEvent) {
  let eventPrepend = "typhonjs";

  const options = ev.pluginOptions;

  // Apply any plugin options.
  if (typeof options === "object") {
    // If `eventPrepend` is defined then it is prepended before all event bindings.
    if (typeof options.eventPrepend === "string") {
      eventPrepend = `${options.eventPrepend}:`;
    }
  }

  ev.eventbus?.on(`${eventPrepend}:ast:walker:traverse`, walker.traverse);
}
