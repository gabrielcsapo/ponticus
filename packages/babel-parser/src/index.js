import BabelParser, { onPluginLoad } from "./BabelParser.js";

export { BabelParser, onPluginLoad };
