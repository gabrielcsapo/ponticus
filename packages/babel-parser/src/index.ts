import BabelParser, { onPluginLoad } from "@ponticus/babel-parser/BabelParser";

export { BabelParser, onPluginLoad };
