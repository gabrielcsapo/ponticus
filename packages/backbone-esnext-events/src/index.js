import EventProxy from "./EventProxy.js";
import TyphonEvents from "./TyphonEvents.js";
import Events from "./Events.js";

export { EventProxy, TyphonEvents, Events };
