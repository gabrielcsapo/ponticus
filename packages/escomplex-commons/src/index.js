import ModuleReport from "./module/report/ModuleReport.js";
import ProjectReport from "./project/report/ProjectReport.js";
import AbstractSyntaxLoader from "./module/plugin/syntax/AbstractSyntaxLoader.js";
import ASTGenerator from "./utils/ast/ASTGenerator.js";
import HalsteadArray from "./module/traits/HalsteadArray.js";
import TraitUtil from "./module/traits/TraitUtil.js";
import actualize from "./module/traits/actualize.js";
import ObjectUtil from "./utils/ObjectUtil.js";
import ModuleScopeControl from "./module/report/control/ModuleScopeControl.js";
import MathUtil from "./utils/MathUtil.js";

export {
  actualize,
  ObjectUtil,
  TraitUtil,
  MathUtil,
  HalsteadArray,
  ASTGenerator,
  ModuleScopeControl,
  AbstractSyntaxLoader,
  ProjectReport,
  ModuleReport,
};
