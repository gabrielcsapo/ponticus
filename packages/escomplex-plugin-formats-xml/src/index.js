import { TransformFormat } from "@ponticus/escomplex-commons";

import FormatXML from "./FormatXML";
import FormatXMLCheckstyle from "./FormatXMLCheckstyle";

/**
 * Iterates through all `json` formats creating a FormatXML instance for each one. `checkstyle` format types need to
 * be transformed and are processed by `FormatXMLCheckstyle`.
 */
TransformFormat.forEachExt("json", (jsonFormat, jsonFormatName) => {
  // Create format name by substituting any leading `json` string for `xml`.
  let formatName = jsonFormatName.replace(/^json/, "xml");

  // If the format name isn't prepended with `xml` then add it.
  if (!formatName.startsWith("xml")) {
    formatName = `xml-${formatName}`;
  }

  switch (jsonFormat.type) {
    case "checkstyle":
      TransformFormat.addFormat(
        new FormatXMLCheckstyle(formatName, jsonFormat.type, jsonFormatName)
      );
      break;

    default:
      TransformFormat.addFormat(
        new FormatXML(formatName, jsonFormat.type, jsonFormatName)
      );
      break;
  }
});
