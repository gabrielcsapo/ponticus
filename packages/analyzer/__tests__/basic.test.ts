import { test, describe, expect } from "vitest";

import { analyse } from "../src/index";

describe("Basic Test", () => {
  test("the test runner should run tests", () => {
    expect(true).toBe(true);
  });

  test("calling analyse has the expected output", () => {
    expect(analyse("const x=5;")).toMatchSnapshot();
  });
});
