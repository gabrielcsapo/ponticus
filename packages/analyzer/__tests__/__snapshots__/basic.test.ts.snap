// Vitest Snapshot v1

exports[`Basic Test > calling analyse has the expected output 1`] = `
ModuleReport {
  "aggregate": AggregateReport {
    "aggregate": undefined,
    "cyclomatic": 1,
    "cyclomaticDensity": 0,
    "halstead": HalsteadData {
      "bugs": 0,
      "difficulty": 0,
      "effort": 0,
      "length": 0,
      "operands": {
        "distinct": 0,
        "identifiers": [],
        "total": 0,
      },
      "operators": {
        "distinct": 0,
        "identifiers": [],
        "total": 0,
      },
      "time": 0,
      "vocabulary": 0,
      "volume": 0,
    },
    "paramCount": 0,
    "sloc": {
      "logical": 0,
      "physical": 1,
    },
  },
  "aggregateAverage": MethodAverage {
    "cyclomatic": 0,
    "cyclomaticDensity": 0,
    "halstead": HalsteadAverage {
      "bugs": 0,
      "difficulty": 0,
      "effort": 0,
      "length": 0,
      "operands": {
        "distinct": 0,
        "total": 0,
      },
      "operators": {
        "distinct": 0,
        "total": 0,
      },
      "time": 0,
      "vocabulary": 0,
      "volume": 0,
    },
    "paramCount": 0,
    "sloc": {
      "logical": 0,
      "physical": 0,
    },
  },
  "classes": [],
  "dependencies": [],
  "errors": [],
  "filePath": undefined,
  "lineEnd": 1,
  "lineStart": 1,
  "maintainability": 0,
  "methodAverage": MethodAverage {
    "cyclomatic": 0,
    "cyclomaticDensity": 0,
    "halstead": HalsteadAverage {
      "bugs": 0,
      "difficulty": 0,
      "effort": 0,
      "length": 0,
      "operands": {
        "distinct": 0,
        "total": 0,
      },
      "operators": {
        "distinct": 0,
        "total": 0,
      },
      "time": 0,
      "vocabulary": 0,
      "volume": 0,
    },
    "paramCount": 0,
    "sloc": {
      "logical": 0,
      "physical": 0,
    },
  },
  "methods": [],
  "settings": {},
  "srcPath": undefined,
  "srcPathAlias": undefined,
}
`;
